lib/client.js

```diff
@@ -416,6 +421,40 @@ Client.prototype.query = function (config, values, callback) {
     }
   }
 
+  if (readTimeout) {
+    queryCallback = query.callback
+
+    readTimeoutTimer = setTimeout(() => {
+      var error = new Error('Query read timeout')
+
+      process.nextTick(() => {
+        query.handleError(error, this.connection)
+      })
+
+      if (typeof queryCallback === 'function') {
+        queryCallback(error)
+
+        // we already returned an error,
+        // just do nothing if query completes
+        query.callback = () => {
+        }
+      }
+
+      // Remove from queue
+      var index = this.queryQueue.indexOf(query)
+      if (index > -1) {
+        this.queryQueue.splice(index, 1)
+      }
+
+      this._pulseQueryQueue()
+    }, readTimeout)
+
+    query.callback = (err, res) => {
+      clearTimeout(readTimeoutTimer)
+      queryCallback(err, res)
```

Comment:
Not sure, but should there be a check for `typeof queryCallback` before making this call (seeing as there was one on line 434)?

Reply:
Yes, I don't think it's needed after all. I've removed it.